week4/weather.py

```diff
@@ -0,0 +1,123 @@
+import requests
+import matplotlib.pyplot as plt
+import random
+
+
+class AbstractDataService:
+    def get_data(self):
+        raise NotImplementedError("Subclasses must implement this method")
+
+class MockedDataService(AbstractDataService):
+    def get_data(self):
+        # Mocked data retrieval logic
+        mocked_data = {
+            'time': [],
+            'temperature_2m': []
+        }
+
+        start_temperature = 5.0
+        end_temperature = 20.0
+
+        # Generate mocked time and temperature data
+        for i in range(72):  # Assuming 72 data points based on the provided results
+            time = f'2023-11-26T{str(i).zfill(2)}:00'
+            temperature = round(random.uniform(start_temperature, end_temperature), 1)
+
+            mocked_data['time'].append(time)
+            mocked_data['temperature_2m'].append(temperature)
```

Comment:
it would be better to not hard-code parameters like `time` into the function, but make it part of the method parameters